src/coreclr/System.Private.CoreLib/src/System/Environment.OSX.CoreCLR.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace System
+{
+    public static partial class Environment
+    {
+#if FEATURE_OBJCWRAPPERS
+        [DllImport(RuntimeHelpers.QCall)]
+        private static extern void GetOperatingSystemVersionFromNSProcessInfo(out int major, out int minor, out int patch);
+
+        private static Version GetMacOSVersion()
+        {
+            GetOperatingSystemVersionFromNSProcessInfo(out int major, out int minor, out int patch);
+            return new Version(major, minor, patch);
```

Comment:
Does this end up being different from what we return from Environment.OSVersion?  On macOS that's using:
https://github.com/dotnet/runtime/blob/1d9e50cb4735df46d3de0cee5791e97295eaf588/src/libraries/Common/src/Interop/OSX/Interop.libobjc.cs#L34-L67

Reply:
Oh, nevermind, I see you're changing the implementation to use this.